galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyQueryRoute.java

```diff
@@ -0,0 +1,135 @@
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+public class PropertyQueryRoute extends CPSRoute{
+
+    private static final String path = "/cps/(.*)/properties([?]?|[^/])+$";
+
+    public PropertyQueryRoute(ResponseBuilder responseBuilder, IFramework framework) {
+        super(responseBuilder, path , framework);
+    }
+
+   
+    @Override
+    public HttpServletResponse handleRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response)
+            throws ServletException, IOException, FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String properties = getNamespaceProperties(namespace, queryParams);
+        return getResponseBuilder().buildResponse(response, "application/json", properties, HttpServletResponse.SC_OK); 
+    }
+
+    private String getNamespaceProperties(String namespace, QueryParameters queryParams) throws FrameworkException{
+        String properties = "";
+         try {
+            nameValidator.assertNamespaceCharPatternIsValid(namespace);
+            if (hiddenNameSpaces.contains(namespace)) {
+            ServletError error = new ServletError(GAL5016_CPS_HIDDEN_NAMESPACE_ERROR, namespace);
+			throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+            String prefix = queryParams.getSingleString("prefix", null);
+            String suffix = queryParams.getSingleString("suffix", null);
+            properties = getProperties(namespace, prefix, suffix);
+        }catch (FrameworkException f){
+            if (f instanceof InternalServletException){
+                throw f;
+            }else{
+                ServletError error = new ServletError(GAL5017_INVALID_NAMESPACE_ERROR,namespace);  
+                throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+        }
+        return properties;
+    }
+    
+    
+    private String getProperties(String namespace, String prefix, String suffix) throws ConfigurationPropertyStoreException {
+        Map<String, String> properties = getAllProperties(namespace);
+       
+        if (prefix != null){
+            properties = filterPropertiesByPrefix(properties,prefix);
+        }
+        if (suffix != null){
+            properties = filterPropertiesBySuffix(properties,suffix);
+        }
+        
+        JsonArray propertyArray = new JsonArray();
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            JsonObject cpsProp = new JsonObject();
+            cpsProp.addProperty("name", entry.getKey());
+            cpsProp.addProperty("value", getProtectedValue(entry.getValue(),namespace));
+            propertyArray.add(cpsProp);
+        }
+        return gson.toJson(propertyArray);
+    }
+    
+    protected  Map<String, String> filterPropertiesByPrefix( Map<String, String> properties , String prefix){
```

Comment:
I think there are some discussions to be had about property resolution, as the prefix and suffix search criteria are far more weird than you might expect.
James D explained it to me and I had a go at writing it down... looking for link.

Reply:
https://pages.github.ibm.com/galasa/developer-docs/500-design/properties/

Reply:
So I think there is a difference between 'searching for properties' and 'resolving a property' using a --prefix and --suffix perhaps.

Probably more discussion required.

Reply:
Added namespace as at the start of the prefix to match the existing behaviour